### ISSUE TITLE
AWS S3: Default credentials

### ISSUE THREAD

**Issue by tylergets** [cshum/imagor#232]
Hey there, I am a bit fan of this project thus far. Thank you for the work.

I am attempting to run imagor on a fargate cluster, and I would like to use the ECS Task Role to handle the S3 permissions so that I don't need to manage a set of keys.

It appears imagor wont attempt to initialize s3 unless the credentials are provided, which stops the aws sdk from trying to default credential provider chain.

Creating this issue to see if theres any potential workaround, or simply bring this to attention.

Thank you!

Heres some documentation

https://docs.aws.amazon.com/sdk-for-go/api/aws/defaults/#CredChain
https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html

**Comment by cshum** [comment]
Hi @tylergets the master branch docker build https://github.com/cshum/imagor/pkgs/container/imagor/48622739?tag=master
```
ghcr.io/cshum/imagor@sha256:6067713f2fd8ffcdd58fee6a56de283b9ad875820d303475b1779ef2493462e6
```
attempts to add support for AWS Default credentials. Would you test out this docker build?

**Comment by alihen** [comment]
I'm busy testing this change - I'm unsure if it's working as expected (more context below)

I was about to open another issue, but posting on this one seems more appropriate:

I'm looking to use the Shared Credentials approach. (e.g $HOME/.aws/credentials), as outlined here - https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html

```
sess, err := session.NewSession(&aws.Config{
    Region:      aws.String("eu-west-1"),
    Credentials: credentials.NewSharedCredentials(""),
})
```

There's quite a bit of overlap between the session and shared credentials approach - it would be great if it could support both, and then continue and look for environment variable access keys/secrets if set for loader/storage/results.

**Comment by tylergets** [comment]
Hey @cshum , thank you for taking a look at this. 

Maybe I am doing something wrong here, however with that build and just the S3_LOADER_BUCKET variable I am not seeing imagor attempt requests to s3; requesting an image such as 

https://********.cloudfront.net/unsafe/test.jpg

fails with 

```{
  "message": "dial tcp: lookup test.jpg on 10.0.0.2:53: no such host: https://test.jpg",
  "status": 404
}```

I took a look at your change and nothing stood out as wrong, so I am going to continue investigating

**Comment by cshum** [comment]
@alihen @tylergets test the following docker build?
https://github.com/cshum/imagor/pkgs/container/imagor/48808278?tag=master
```
ghcr.io/cshum/imagor@sha256:a6cfa229fb220c1836eade62fe5254bde13163acbdda8aa4c8ab06270e8c9162
```

**Comment by tylergets** [comment]
@cshum that did it, works great. Thank you!

**Comment by cshum** [comment]
Available in v1.2.3
